### PR TITLE
Update to Compose SNAPSHOT 7094152

### PR DESCRIPTION
**Pull request by chrisbanes**
I still need to migrate away from the old/deprecated `transition` API, but I'll follow up on that when I find some time.

### CODE REVIEW

**Comment by chrisbanes** [comment]
I think this is finally ready for review. This PR was a nightmare.

File: coil/src/androidTest/java/dev/chrisbanes/accompanist/coil/CoilTest.kt [hunk 1]
```diff
@@ -75,8 +76,9 @@ import java.util.concurrent.atomic.AtomicInteger
 @LargeTest
 @RunWith(JUnit4::class)
 class CoilTest {
+    @Suppress("DEPRECATION") // createAndroidComposeRuleLegacy
     @get:Rule
-    val composeTestRule = createComposeRule()
+    val composeTestRule = androidx.compose.ui.test.junit4.createAndroidComposeRuleLegacy<ComponentActivity>()
```
Comment (nickbutcher) [inline coil/src/androidTest/java/dev/chrisbanes/accompanist/coil/CoilTest.kt:81]: Add a comment why legacy is needed?

File: glide/src/androidTest/java/dev/chrisbanes/accompanist/glide/GlideTest.kt [hunk 2]
```diff
@@ -74,8 +74,9 @@ import java.util.concurrent.TimeUnit
 @LargeTest
 @RunWith(JUnit4::class)
 class GlideTest {
+    @Suppress("DEPRECATION") // createAndroidComposeRuleLegacy
     @get:Rule
-    val composeTestRule = createComposeRule()
+    val composeTestRule = androidx.compose.ui.test.junit4.createAndroidComposeRuleLegacy<ComponentActivity>()
```
Comment (nickbutcher) [inline glide/src/androidTest/java/dev/chrisbanes/accompanist/glide/GlideTest.kt:79]: ditto comment

File: coil/src/androidTest/java/dev/chrisbanes/accompanist/coil/CoilTest.kt (hunk 1 above)
Comment (chrisbanes) [reply coil/src/androidTest/java/dev/chrisbanes/accompanist/coil/CoilTest.kt:81]: Not sure it's worth it. I hope to get the new stuff landed in #186. I just need to unblocked this PR.

**Review by JoseAlcerreca — APPROVED** [review]

**Pull request merged**